com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,225 @@
+            when (ad) {
+                is HeliumFullscreenAd -> asFullScreen()?.readyToShow() ?: false
+                is HeliumBannerAd -> {
+                    Log.w(TAG, "This should never be called, banners do not have readyToShow")
+                    false
+                }
+                else -> false
+            }
+        } catch (ex: Exception) {
+            Log.w(TAG, "Helium encountered an error calling Ad.readyToShow() - ${ex.message}")
+            false
+        }
+    }
+
+    fun destroy() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            destroyBannerLayout()
+            ad.destroy()
+        }
+    }
+
+    private fun createBannerLayout(screenLocation: Int) {
+        if (ad !is HeliumBannerAd)
+            return
+
+        bannerLayout?.let {
+            it.removeAllViews()
+        }
+
+        // Create the banner layout on the given position.
+        // Check if there is an already existing banner layout. If so, remove it. Otherwise,
+        // create a new one.
+        bannerLayout?.let {
+            val bannerParent = it.parent as FrameLayout
+            bannerParent.removeView(it)
+        } ?: run {
+            bannerLayout = RelativeLayout(activity)
+            bannerLayout?.setBackgroundColor(Color.TRANSPARENT)
+        }
+
+        /*
+            //     TopLeft = 0,
+            //     TopCenter = 1,
+            //     TopRight = 2,
+            //     Center = 3,
+            //     BottomLeft = 4,
+            //     BottomCenter = 5,
+            //     BottomRight = 6
+        */
+        var bannerGravityPosition = 0
+        when (screenLocation) {
+            0 -> bannerGravityPosition = Gravity.TOP or Gravity.LEFT
+            1 -> bannerGravityPosition = Gravity.TOP or Gravity.CENTER_HORIZONTAL
+            2 -> bannerGravityPosition = Gravity.TOP or Gravity.RIGHT
+            3 -> bannerGravityPosition = Gravity.CENTER_VERTICAL or Gravity.CENTER_HORIZONTAL
+            4 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.LEFT
+            5 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.CENTER_HORIZONTAL
+            6 -> bannerGravityPosition = Gravity.BOTTOM or Gravity.RIGHT
+        }
+        bannerLayout?.gravity = bannerGravityPosition
+
+        // Attach the banner layout to the activity.
+        val pixels = displayDensity
+        try {
+            val bannerAd = asBanner()
+            when (if (bannerAd?.getSize() != null) bannerAd.getSize() else HeliumBannerSize.STANDARD) {
+                HeliumBannerSize.LEADERBOARD -> bannerAd?.layoutParams = getBannerLayoutParams(pixels, LEADERBOARD_WIDTH, LEADERBOARD_HEIGHT)
+                HeliumBannerSize.MEDIUM -> bannerAd?.layoutParams = getBannerLayoutParams(pixels, MEDIUM_WIDTH, MEDIUM_HEIGHT)
+                HeliumBannerSize.STANDARD -> bannerAd?.layoutParams = getBannerLayoutParams(pixels, STANDARD_WIDTH, STANDARD_HEIGHT)
+                null -> TODO()
+            }
```

Comment:
I think having `if else` inside of `when` is not common and makes it somewhat harder to follow. Could we separate them?

Reply:
Cleaned it up